openshift/vault-manager.template.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: vault-manager
+objects:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    labels:
+      app: vault-manager
+    name: vault-manager
+  spec:
+    replicas: ${{REPLICAS}}
+    strategy:
+      type: RollingUpdate
+      rollingUpdate:
+        maxUnavailable: 0
+        maxSurge: 1
+    selector:
+      matchLabels:
+        app: vault-manager
+    template:
+      metadata:
+        labels:
+          app: vault-manager
+      spec:
+        containers:
+        - image: ${IMAGE}:${IMAGE_TAG}
+          imagePullPolicy: Always
+          name: vault-manager
+          command: ["/bin/sh"]
+          args: 
+          - "-c"
+          - "while true; do /vault-manager {{ if ${DRY_RUN} == 'True' }}--dry-run{{ end }}; sleep ${RECONCILE_SLEEP_TIME}; done;"
+          env:
+          - name: GRAPHQL_QUERY_FILE
+            value: ${GRAPHQL_QUERY_FILE}
+          - name: GRAPHQL_SERVER_URL
+            valueFrom:
+              secretKeyRef:
+                key: gql_base_url
+                name: ${VAULT_MANAGER}
+          - name: GRAPHQL_USERNAME
+            valueFrom:
+              secretKeyRef:
+                key: gql_username
+                name: ${VAULT_MANAGER}
+          - name: GRAPHQL_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                key: gql_password
+                name: ${VAULT_MANAGER}
+          - name: VAULT_ADDR
+            value: ${VAULT_ADDR}
+          - name: VAULT_AUTHTYPE
+            valueFrom:
+              secretKeyRef:
+                key: vault_auth_type
+                name: ${VAULT_MANAGER}
+          - name: VAULT_ROLE_ID
+            valueFrom:
+              secretKeyRef:
+                key: vault_role_id
+                name: ${VAULT_MANAGER}
+          - name: VAULT_SECRET_ID
+            secretKeyRef:
+                key: vault_secret_id
+                name: ${VAULT_MANAGER}
+          resources:
+            requests:
+              memory: ${MEMORY_REQUESTS}
+              cpu: ${CPU_REQUESTS}
+            limits:
+              memory: ${MEMORY_LIMIT}
+              cpu: ${CPU_LIMIT}
+parameters:
+- name: IMAGE
+  value: quay.io/app-sre/vault-manager
+  displayName: vault-manager image
+  description: vault-manager docker image. Defaults to quay.io/app-sre/vault-manager
+- name: IMAGE_TAG
+  value: latest
+  displayName: vault-manager version
+  description: vault-manager version which defaults to latest
+- name: MEMORY_REQUESTS
+  value: 200Mi
+- name: MEMORY_LIMIT
```

Comment:
I bumped this up for now. docker stats on the running container didn't show much in terms of memory or cpu usage. There are some deeper profiling methods I can get into.

What's the thought on dialing these experimentally via kube container metrics on a staging deployment?

Reply:
i think it's ok. you can also look at https://blog.golang.org/pprof.